### PR TITLE
[Testing] Remove the data races in zerolog Hook

### PR DESCRIPTION
**Pull request by SaveTheRbtz**

### CODE REVIEW

File: utils/unittest/logging.go [hunk 1]
```diff
@@ -59,16 +60,22 @@ func HookedLogger() (zerolog.Logger, LoggerHook) {
...
+	defer hook.mu.Unlock()
+
 	return hook.logs.String()
 }
 
 // Run implements zerolog.Hook and appends the log message to the log.
 func (hook LoggerHook) Run(_ *zerolog.Event, level zerolog.Level, msg string) {
+	hook.mu.Lock()
 	hook.logs.WriteString(msg)
+	hook.mu.Unlock()
```
Comment (pattyshack) [inline utils/unittest/logging.go:78]: use defer unlock pattern plz

**Review by pattyshack — APPROVED** [review]

File: utils/unittest/logging.go (hunk 1 above)
Comment (SaveTheRbtz) [reply utils/unittest/logging.go:78]: I did not want to hold the lock during `Println`.... let's see, maybe I'll just factor it out into a separate function (it'll be inlined anyway)

Comment (pattyshack) [reply utils/unittest/logging.go:78]: maybe just print the line before writing to logs?

**Review by jordanschalm — APPROVED** [review]

**Comment by SaveTheRbtz** [comment]
bors merge

**Pull request merged**